### ISSUE TITLE
Duplicate lint failures for issues in subcharts when using `--with-subcharts`

### ISSUE THREAD

**Issue by allanlewis** [helm/helm#8488]
I have a chart, let's call it "parent", that contains a subchart, let's call it "child", under `charts/`. The subchart causes a linting error but this is reported twice, as per the following redacted output:

```
$ helm lint --strict --with-subcharts ./parent
==> Linting ./parent
[INFO] Chart.yaml: icon is recommended
[ERROR] templates/: template: parent/charts/child/templates/<file>:<line>:<char>: executing "parent/charts/child/templates/<file>" at <$.Values.foo.bar>: map has no entry for key "bar"

==> Linting parent/charts/child
[INFO] Chart.yaml: icon is recommended
[ERROR] templates/: template: child/templates/<file>:<line>:<char>: executing "child/templates/<file>" at <$.Values.foo.bar>: map has no entry for key "bar"

Error: 2 chart(s) linted, 2 chart(s) failed
```

The same error is reported twice. Given that the parent chart has no linting issues in itself, I think this error should only be reported for the subchart.

If I remove the `--with-subcharts` flag, I still get the error for the subchart reported under the parent chart, so I guess the workaround is not to use `--with-subcharts` at all, although this might miss other issues in the subchart.

> Output of `helm version`:
> ```
> version.BuildInfo{Version:"v3.2.4", GitCommit:"0ad800ef43d3b826f31a5ad8dfbb4fe05d143688", GitTreeState:"clean", GoVersion:"go1.13.12"}
> ```
> Output of `kubectl version`:
> ```
> Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.12", GitCommit:"a8b52209ee172232b6db7a6e0ce2adc77458829f", GitTreeState:"clean", BuildDate:"2019-10-15T12:12:15Z", GoVersion:"go1.11.13", Compiler:"gc", Platform:"linux/amd64"}
> Server Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.7", GitCommit:"6f482974b76db3f1e0f5d24605a9d1d38fad9a2b", GitTreeState:"clean", BuildDate:"2019-03-25T02:41:57Z", GoVersion:"go1.10.8", Compiler:"gc", Platform:"linux/amd64"}
> ```
> Cloud Provider/Platform (AKS, GKE, Minikube etc.): n/a

**Comment by allanlewis** [comment]
There might be something more going on here. I've noticed that I have precisely the same issue in the same file in my parent chart as in my subchart; it's using a value provided by `helm create` that I've deleted, but there's still a reference in the template. (The file in question is actually `NOTES.txt` in both charts, in case that's handled differently.)

If I add some lines to `NOTES.txt` in the parent chart, the line number reported by `helm lint` is still the line from the child's `NOTES.txt`, i.e. `helm lint` _isn't reporting_ the error in the parent chart. I'm not sure if this is part of the same issue, or a separate issue.
